java/76_Minimum_Window_Substring.java

```diff
@@ -0,0 +1,64 @@
+//76. Minimum Window Substring
+/**
+ * Given a string S and a string T, find the minimum window in S which will contain all the characters in T in complexity O(n).
+ * 
+ * For example,
+ * S = "ADOBECODEBANC"
+ * T = "ABC"
+ * Minimum window is "BANC".
+ * 
+ * Note:
+ * If there is no such window in S that covers all characters in T, return the empty string "".
+ * 
+ * If there are multiple such windows, you are guaranteed that there will always be only one unique minimum window in S.
+ * 
+ * Tags: Hash Table, Two Pointers, String
+ *
+ * Similar Problems: (H) Substring with Concatenation of All Words, (M) Minimum Size Subarray Sum, (H) Sliding Window Maximum
+ *
+ * Author: Xinyue Zhang
+ */
+ 
+ public class Solution {
+    public String minWindow(String s, String t) {
+        if(s.isEmpty() || t.isEmpty()) 
+            return "";
+        
+        HashMap<Character,Integer> map = new HashMap<Character,Integer>();
```

Comment:
Actually here you can use an array `int map[128]` instead of a hashmap structure. Because the ascii of a character can be use as the index. For example, the ascii of 'A' is 65, so map['A'] is equivalent to map[65]. It can be used as a natural hashmap, then you can get rid of a lot of complex operations below.